src/load_dotenv.rs

```diff
@@ -1,45 +1,65 @@
 use crate::common::*;
 
+const DEFAULT_DOTENV_FILENAME: &str = ".env";
+
 pub(crate) fn load_dotenv(
   config: &Config,
   settings: &Settings,
   working_directory: &Path,
 ) -> RunResult<'static, BTreeMap<String, String>> {
-  // `dotenv::from_path_iter` should eventually be un-deprecated, see:
-  // https://github.com/dotenv-rs/dotenv/issues/13
-  #![allow(deprecated)]
-
   if !settings.dotenv_load.unwrap_or(true) {
     return Ok(BTreeMap::new());
   }
 
-  for directory in working_directory.ancestors() {
-    let path = directory.join(".env");
+  if let Some(path) = &config.dotenv_path {
+    return load_from_file(config, settings, &path);
+  }
+
+  let filename = match &config.dotenv_filename {
+    Some(name) => name,
+    None => DEFAULT_DOTENV_FILENAME,
+  };
```

Comment:
Can this be something like:

```suggestion
  let filename = config.dotenv_filename.unwrap_or(DEFAULT_DOTENV_FILENAME);
```

Might need to juggle the types a little bit to get it to work.